test/gtest/ucp/test_ucp_ep.cc

```diff
@@ -62,4 +61,37 @@ UCS_TEST_P(test_ucp_ep, ucp_query_ep)
                     estimated_time_1000 - estimated_time_0);
 }
 
+UCS_TEST_P(test_ucp_ep, ucp_query_transport)
+{
+    ucp_ep_h ep;
+    ucs_status_t status;
+    int i;
+    ucp_ep_attr_t ep_attrs;
+
+    ep                              = sender().ep();
+    ep_attrs.field_mask             = UCP_EP_ATTR_FIELD_TRANSPORTS;
+    ep_attrs.transports.entries     = new ucp_transport_entry_t[10];
+    ep_attrs.transports.num_entries = 100;
+    ep_attrs.transports.entry_size  = sizeof(ucp_transport_entry_t);
+    status                          = ucp_ep_query(ep, &ep_attrs);
+
+    // Verify ucp_ep_query completed successfully, that the number of
+    // transports is updated, since no system should reasonably have
+    // 100 transport/device name pairs, and verify returned strings are
+    // not empty.
+    EXPECT_EQ(status, UCS_OK);
+    EXPECT_LT(ep_attrs.transports.num_entries, 100);
```

Comment:
EXPECT_LE

Reply:
I can change this to EXPECT_LE, but my intent with this test was to prove that the UCX code was updating the num_entries field, since I think there will never be 100 transport/device pairs on a system, which means that the num_entries field must be set to something < 100 on return.